nx/lib/nx.ex

```diff
@@ -6128,6 +6128,13 @@ defmodule Nx do
 
   is true for all elements of a and b.
 
+  ## Options
+
+    * `:rtol` - relative tolerance between numbers, as described above. Defaults to 1.0e-5
+    * `:atol` - absolute tolerance between numbers, as described above. Defaults to 1.0e-8
+    * `:equal_nan` - if `false`, NaN will always compare as false.
+      Otherwise `NaN` will only equal `NaN`. Defaults to `true`
```

Comment:
I think the default should be false. This will actually help us detect NaNs in unexpected places. If expected, then you set the flag to true. WDYT?

Reply:
That makes sense! I'll change that and then merge